Ubuntu: reported by Agostino Russo
Upstream: http://lists.gnu.org/archive/html/grub-devel/2009-09/msg00252.html
Description: Fix test crash if loopback device does not contain a filesystem.

diff -Nur -x '*.orig' -x '*~' grub2-1.97~beta3/commands/test.c grub2-1.97~beta3.new/commands/test.c
--- grub2-1.97~beta3/commands/test.c	2009-09-12 14:29:38.000000000 +0100
+++ grub2-1.97~beta3.new/commands/test.c	2009-09-14 10:51:20.000000000 +0100
@@ -88,6 +88,13 @@
       }
 
     fs = grub_fs_probe (dev);
+    if (! fs)
+      {
+	grub_free (device_name);
+	grub_device_close (dev);
+	return;
+      }
+
     pathname = grub_strchr (path, ')');
     if (! pathname)
       pathname = path;
